gqcp/src/ONVBasis/SeniorityZeroONVBasis.cpp

```diff
@@ -55,179 +57,323 @@ size_t SeniorityZeroONVBasis::calculateDimension(const size_t K, const size_t N_
 
-//     // We don't just use the sum of the one- and two-electron operator's diagonal representation because that would mean 2 iterations over the dimension of the ONV basis.
+    // We don't just use the sum of the one- and two-electron operator's diagonal representation because that would mean 2 iterations over the dimension of the ONV basis.
 
 
-//     // Check if the argument is compatible.
-//     const auto K = sq_hamiltonian.numberOfOrbitals();  // number of spatial orbitals
+    // Check if the argument is compatible.
+    const auto K = hamiltonian.numberOfOrbitals();  // number of spatial orbitals
 
-//     if (K != this->numberOfSpatialOrbitals()) {
-//         throw std::invalid_argument("SeniorityZeroONVBasis::evaluateOperatorDiagonal(const RSQHamiltonian<double>&): The number of spatial orbitals for the ONV basis and one-electron operator are incompatible.");
-//     }
+    if (K != this->numberOfSpatialOrbitals()) {
+        throw std::invalid_argument("SeniorityZeroONVBasis::evaluateOperatorDiagonal(const RSQHamiltonian<double>&): The number of spatial orbitals for the ONV basis and one-electron operator are incompatible.");
+    }
 
-//     // Prepare some variables to be used in the algorithm.
-//     const auto dim = this->dimension();
+    // Prepare some variables to be used in the algorithm.
+    const auto dim = this->dimension();
 
-//     const auto& h = sq_hamiltonian.core().parameters();
-//     const auto& g = sq_hamiltonian.twoElectron().parameters();
+    const auto& h = hamiltonian.core().parameters();
+    const auto& g = hamiltonian.twoElectron().parameters();
 
-//     VectorX<double> diagonal = VectorX<double>::Zero(dim);
+    VectorX<double> diagonal = VectorX<double>::Zero(dim);
 
 
-//     // Iterate over every proxy doubly-occupied ONV. Since we are actually using spin-unresolved ONVs, we should multiply contributions by 2.
-//     this->forEach([&diagonal, &h, &g](const SpinUnresolvedONV& onv, const size_t I) {
-//         double value = 0;  // to be added to the diagonal
+    // Iterate over every proxy doubly-occupied ONV. Since we are actually using spin-unresolved ONVs, we should multiply contributions by 2.
+    this->forEach([&diagonal, &h, &g](const SpinUnresolvedONV& onv, const size_t I) {
+        double value = 0;  // to be added to the diagonal
 
-//         // Loop over every occupied spinor index and add the contributions.
-//         onv.forEach([&value, &h, &g](const size_t p) {
-//             value += 2 * h(p, p);    // *2 because of seniority zero
-//             value += g(p, p, p, p);  // 1/2*2 because of seniority zero
-//         });
+        // Loop over every occupied spinor index and add the contributions.
+        onv.forEach([&value, &h, &g](const size_t p) {
+            value += 2 * h(p, p);    // *2 because of seniority zero
+            value += g(p, p, p, p);  // 1/2*2 because of seniority zero
```

Comment:
```suggestion
            value += 2 * h(p, p);    // Factor *2 because of seniority zero.
            value += g(p, p, p, p);  // Factor 1/2*2 because of seniority zero.
```